internal/provider/resource_access_generic.go

```diff
@@ -140,27 +155,290 @@ func (a *genericJAASAccessResource) Configure(ctx context.Context, req resource.
 		)
 		return
 	}
-	a.client = client
+	resource.client = client
 	// Create the local logging subsystem here, using the TF context when creating it.
-	a.subCtx = tflog.NewSubsystem(ctx, a.resourceLogName)
+	resource.subCtx = tflog.NewSubsystem(ctx, resource.resourceLogName)
 }
 
 // Create defines how tuples for access control will be created.
-func (a *genericJAASAccessResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+func (resource *genericJAASAccessResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Check first if the client is configured
+	if resource.client == nil {
+		addClientNotConfiguredError(&resp.Diagnostics, resource.resourceLogName, "create")
+		return
+	}
+
+	// Read Terraform configuration from the request into the model
+	plan, targetTag := resource.info(ctx, req.Plan, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
+	// Create tuples to create from the plan
+	tuples := modelToTuples(ctx, targetTag, plan, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// Make a call to create relations
+	err := resource.client.Jaas.AddRelations(tuples)
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to create access relationships, got error: %s", err))
```

Comment:
suggestions: add the value of the resource type (more targetTag.Id) into the error message. the messages can occur in a parallel flood, it won't necessarily be apparent which resource had the issue if more than one.

Same for all "Client Errors" in this file.

Reply:
Adding the targetTag.String() to the error. But I don't think this is strictly necessary since Terraform errors already include the resource name in the error, an example below.
```
│ Error: Client Error
│ 
│   with module.amf.juju_application.amf,
│   on .terraform/modules/amf/terraform/main.tf line 4, in resource "juju_application" "amf":
│    4: resource "juju_application" "amf" {
│ 
│ Unable to update application resource, got error: updating charm config: cannot upgrade application "amf" to
│ charm "ch:amd64/jammy/sdcore-amf-k8s-669": would break relation "amf:database mongodb:database"
```